tests/test_scf.py

```diff
@@ -0,0 +1,58 @@
+import sys
+import mlir
+from mlir.dialects.func import func 
+
+# All source strings taken from examples in https://mlir.llvm.org/docs/Dialects/SCFDialect/
+
+
+def assert_roundtrip_equivalence(source):
+    assert source == mlir.parse_string(source).dump()
+
+
+def test_scf_for():
+    assert_roundtrip_equivalence("""module {
+  func.func @reduce(%buffer: memref<1024xf32>, %lb: index, %ub: index, %step: index, %sum_0: f32) -> (f32) {
+    %sum = scf.for %iv = %lb to %ub step %step iter_args ( %sum_iter = %sum_0 ) -> ( f32 ) {
+      %t = load %buffer [ %iv ] : memref<1024xf32>
+      %sum_next = arith.addf %sum_iter, %t : f32
+      scf.yield %sum_next : f32
+    }
+    return %sum : f32
+  }
+}""")
+
+
+def test_scf_if():
+    assert_roundtrip_equivalence("""module {
+  func.func @example(%A: f32, %B: f32, %C: f32, %D: f32) {
+    %x, %y = scf.if %b -> ( f32, f32 ) {
+      scf.yield %A, %B : f32, f32
+    } else {
+      scf.yield %C, %D : f32, f32
+    }
+    return
+  }
+}""")
+
+
+def test_scf_while():
+    assert_roundtrip_equivalence("""module {
+  func.func @example(%A: f32, %B: f32, %C: f32, %D: f32) {
+    %res = scf.while ( %arg1 = %init1 ) : (f32) -> f32 {
+      %condition = func.call @evaluate_condition ( %arg1 ) : (f32) -> i1
+      scf.condition ( %condition ) %arg1 : f32
+    } do {
+      ^bb0 (%arg2: f32):
+        %next = func.call @payload ( %arg2 ) : (f32) -> f32
+        scf.yield %next : f32
+    }
+  }
+}""")
+
+
+if __name__ == "__main__":
+    if len(sys.argv) > 1:
+        exec(sys.argv[1])
+    else:
+        from pytest import main
+        main([__file__])
```

Comment:
```suggestion
    from pytest import main
    main([__file__])
```
Please don't add `exec` calls into the code (even if it's a main function of a test file). Thanks!

Reply:
Copied this from test_linalg.py, so I'll make the change there too.

Reply:
And test_builder.py